### PR TITLE
Remove terminal specific command from e2e java test file

### PR DESCRIPTION
**Pull request by amitkrout**
## What is the purpose of this change? What does it change?

Update the test file to run it on all oc supported platform including windows
## Was the change discussed in an issue?
fixes Part of #1129

## How to test changes?

make test-java-e2e

### CODE REVIEW

File: tests/e2e/utils.go [hunk 1]
```diff
@@ -90,3 +97,61 @@ func getActiveApplication() string {
 	odoActiveApp := strings.Split(reActiveApp.FindString(stdOut), "*")[1]
 	return strings.TrimSpace(odoActiveApp)
 }
+
+func retryingForOutputMatchStringOfHTTPResponse(url, match string, retry, sleep int) bool {
+	for i := 0; i < retry; i++ {
```
Comment (syamgk) [inline tests/e2e/utils.go]: same here, use simple for loop instead of initializing variable

Comment (amitkrout) [reply tests/e2e/utils.go]: ```retry``` is not an array of string. so can not be operated through a range.

File: tests/e2e/utils.go [hunk 2]
```diff
@@ -126,3 +126,51 @@ func matchResponseSubString(url, match string, retry, sleep int) bool {
 	fmt.Printf("Could not get the match string \"%s\" in %d seconds\n", match, i)
 	return false
 }
+
+// this function returns build name of a delopyed component by passing
+// component name as a argument
+func getBuildNameUsingOc(compName string) string {
```
Comment (kadel) [inline tests/e2e/utils.go]: is`UsingOc` in every function really necessary?

Comment (amitkrout) [reply tests/e2e/utils.go]: Yah, not really. 

Actually the ```oc``` command is being used to fetch the data. So i made the function name in that way. Also it looks more specific that we are executing using ```oc``` not ```odo```. IMO the function name can be more logical by changing the name like ```ocGetBuildName``` WDYT ?

**Comment by amitkrout** [comment]
@kadel I have updated the function name. Please revisit the pr

File: tests/e2e/utils.go (hunk 2 above)
Comment (kadel) [reply tests/e2e/utils.go]: You don't need to mention oc at all. It is clear that it is using oc and not odo. There are no builds in odo. The longer name function has the harder it is to read the code. 

If you have long name like:
```
getBuildNameUsingOc("component")
```

it takes more efford to read it compared to something like this:
```
getBuildName("component")
```


The fact that it is using oc can be mentioned in function description (comment) not in a name

Comment (amitkrout) [reply tests/e2e/utils.go]: hmm, make sense to me 

Thanks for the feedback

Comment (amitkrout) [reply tests/e2e/utils.go]: Done

**Comment by amitkrout** [comment]
Rebased

**Comment by kadel** [comment]
ping @syamgk

File: tests/e2e/java_test.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ var _ = Describe("odoJavaE2e", func() {
 			// Deploy the git repo / wildfly example
 			cmpCreateLog := runCmdShouldPass("odo create wildfly javaee-git-test --git " + warGitRepo + " -w")
 			Expect(cmpCreateLog).ShouldNot(ContainSubstring("This may take a few moments to be ready"))
-			buildName := runCmdShouldPass("oc get builds --output='name' | grep javaee-git-test | cut -d '/' -f 2")
+			buildName := getBuildName("javaee-git-test")
```
Comment (mohammedzee1000) [inline tests/e2e/java_test.go]: +1

File: tests/e2e/java_test.go [hunk 4]
```diff
@@ -55,8 +55,8 @@ var _ = Describe("odoJavaE2e", func() {
 			routeURL := determineRouteURL()
 
 			// Ping said URL
-			pingValue := waitForEqualCmd("curl -s "+routeURL+" | grep 'Insult' | wc -l | tr -d '\n'", "1", 10)
-			Expect(pingValue).To(Equal(true))
+			responsePing := matchResponseSubString(routeURL, "Insult", 90, 1)
```
Comment (mohammedzee1000) [inline tests/e2e/java_test.go]: +1

File: tests/e2e/java_test.go [hunk 5]
```diff
@@ -66,18 +66,17 @@ var _ = Describe("odoJavaE2e", func() {
...
-			buildName := runCmdShouldPass("oc get builds --output='name' | grep wo-wait-javaee-git-test | cut -d '/' -f 2")
+			buildName := getBuildName("wo-wait-javaee-git-test")
 			Expect(buildName).To(ContainSubstring("wo-wait-javaee-git-test"))
-
-			buildStatus := runCmdShouldPass("oc get builds | grep wo-wait-javaee-git-test")
+			buildStatus := getBuildParameterValues("wo-wait-javaee-git-test")
 			Expect(buildStatus).To(ContainSubstring("Pending"))
 
-			dcName := runCmdShouldPass("oc get dc | grep wo-wait-javaee-git-test | cut -f 1 -d ' '")
+			dcName := getDcName("wo-wait-javaee-git-test")
```
Comment (mohammedzee1000) [inline tests/e2e/java_test.go]: +1

File: tests/e2e/java_test.go [hunk 6]
```diff
@@ -66,18 +66,17 @@ var _ = Describe("odoJavaE2e", func() {
...
 			// For waiting until the deployment starts
 			for {
 				time.Sleep(5 * time.Second)
-				dcStatus := runCmdShouldPass("oc get dc |grep wo-wait-javaee-git-test | cut -f 4 -d ' '")
-				if dcStatus == "1\n" {
+				dcStatus := getDcStatusValue("wo-wait-javaee-git-test")
```
Comment (mohammedzee1000) [inline tests/e2e/java_test.go]: +1

**Pull request merged**